File: frontend/src/app/models/izdate.ts
import { Knjiga } from "./knjiga";
import { Osoba } from "./osoba";

export interface Izdate{
  id?:number
  knjigaId: number;
  userId: number;
  ime: string;
  prezime: string;
  trajan: boolean;
  datumVracanja?: string | null; // ISO format datuma ili null
}

File: backend/src/main/java/org/projekat/model/Izdate.java
package org.projekat.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.projekat.model.Osoba;

import java.time.LocalDate;

@AllArgsConstructor
@NoArgsConstructor
@Data
@Entity
public class Izdate {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    @ManyToOne
    @JoinColumn(name = "osoba_id",referencedColumnName = "user_id")
    private Osoba osoba;
    @ManyToOne
    @JoinColumn(name = "knjiga_id",referencedColumnName = "id")
    private Knjiga knjiga;
    private LocalDate datumIzdavanja;
    private LocalDate datumVracanja;
}
